File: tamper-scripts/ruby/xforwarded.rb
#!/usr/bin/env ruby
#
# Author:       KING SABRI | @KINGSABRI
# Description:  Append a fake HTTP header 'X-Forwarded-For' to bypass
#               WAF (usually application based) protection
# Requirements: None
#
require 'json'

@json    = JSON.parse(ARGV[0])
@payload = @json["payload"]
@kwargs  = @json["kwargs"]

@json["kwargs"]["headers"]["X-Forwarded-For"] = Array.new(4){rand(256)}.join('.')

print @json.to_json

File: tamper-scripts/php/xforwarded.php
#!/usr/bin/env php
<?php
#
# Author:       KING SABRI | @KINGSABRI
# Description:  Append a fake HTTP header 'X-Forwarded-For' to bypass
#               WAF (usually application based) protection
# Requirements: None
#
$json    = json_decode($argv[1], true);
$payload = $json['payload'];
$kwargs  = $json['kwargs'];

$json["kwargs"]["headers"]["X-Forwarded-For"] = long2ip(mt_rand());

echo json_encode($json);
?>
